### PR TITLE
feature/JM-6254: Manage court room locations

### PR DESCRIPTION
**Pull request by callum-r-young**
### JIRA link (if applicable) ###

[JM-6254](https://centralgovernmentcgi.atlassian.net/browse/JM-6254)

### Change description ###

Added functionality for Court Manager's to view, add, and edit court room locations at their court.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: server/routes/administration/room-locations/room-locations.controller.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+  const _ = require('lodash');
+  const { validate } = require('validate.js');
+  const { courtroomsDAO } = require('../../../objects/administration');
+  const validator = require('../../../config/validation/create-courtroom');
+  const { replaceAllObjKeys } = require('../../../lib/mod-utils');
+
+  module.exports.getRooms = function(app) {
+    return (req, res) => {
+      return res.redirect(app.namedRoutes.build('administration.room-locations.location.get', {
+        locationCode: req.session.authentication.owner,
```
Comment (ricardofreitasrocha) [inline server/routes/administration/room-locations/room-locations.controller.js:12]: this needs to use a different value. I just spoke with ben and the owner will be the wrong value to use. ping me to pass you the correct ways 😄

File: server/config/validation/create-courtroom.spec.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  'use strict';
+  var validate = require('validate.js')
+    , validator = require('./create-courtroom')
+    , validatorResult = null;
+
+  describe('Administration - Create courtroom validators:', function() {
+
+    beforeEach(function() {
+      validatorResult = null;
+    });
```
Comment (ricardofreitasrocha) [inline server/config/validation/create-courtroom.spec.js:11]: we can now just scope the `validatorResult` to each test case.

```js
it('should test something', function() {
  // ... some code ...

  const validatorResult = validate();

  // ... more code ...
});
```

File: server/routes/administration/room-locations/room-locations.controller.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  const validator = require('../../../config/validation/create-courtroom');
+  const { replaceAllObjKeys } = require('../../../lib/mod-utils');
+
+  module.exports.getRooms = function(app) {
+    return (req, res) => {
+      return res.redirect(app.namedRoutes.build('administration.room-locations.location.get', {
+        locationCode: req.session.authentication.locCode,
+      }));
+    };
+  };
```
Comment (ricardofreitasrocha) [inline server/routes/administration/room-locations/room-locations.controller.js:15]: Why use this to redirect to the `getRoomLocations` controller function and not call it directly? Is it just for handling the locationCode?

Comment (callum-r-young) [reply server/routes/administration/room-locations/room-locations.controller.js:15]: @ricardofreitasrocha Currently i had it for when a court user clicks on the link in the nav bar, i dont think theres a way for us to access the location code directly in nunjucks without first passing it in, and i dont want to have to pass it in everytime we render something that uses court-admin nav.
Open to suggestions though?

File: server/config/validation/create-courtroom.spec.js (hunk 2 above)
Comment (callum-r-young) [reply server/config/validation/create-courtroom.spec.js]: Updated.

File: server/routes/administration/room-locations/room-locations.controller.js (hunk 3 above)
Comment (ricardofreitasrocha) [reply server/routes/administration/room-locations/room-locations.controller.js:15]: that works for me, was just a general question 😄

**Review by ricardofreitasrocha — APPROVED** [review]

**Pull request merged**